llpc/builder/llpcBuilderImpl.h

```diff
@@ -438,6 +438,169 @@ class BuilderImplImage : virtual public BuilderImplBase
     LLPC_DISALLOW_DEFAULT_CTOR(BuilderImplImage)
     LLPC_DISALLOW_COPY_AND_ASSIGN(BuilderImplImage)
 
+    // Represents the type of sampler filter.
+    enum class SamplerFilter : uint32_t
+    {
+        Nearest = 0,
+        Linear,
+    };
+
+    // Represents the YCbCr conversion model.
+    enum class SamplerYCbCrModelConversion : uint32_t
+    {
+        RgbIdentity = 0,
+        YCbCrIdentity,
+        YCbCr709,
+        YCbCr601,
+        YCbCr2020,
+    };
+
+    // Represents whether color channels are encoded using the full range of numerical
+    // values or whether values are reserved for headroom and foot room.
+    enum class SamplerYCbCrRange : uint32_t
+    {
+        ItuFull = 0,
+        ItuNarrow,
+    };
+
+    // Represents the location of downsampled chroma channel samples relative to the luma samples.
+    enum class ChromaLocation : uint32_t
+    {
+        CositedEven = 0,
+        Midpoint,
+    };
+
+    // Represents the component values placed in each component of the output vector.
+    class ComponentSwizzle
+    {
+    public:
+        enum Value: uint32_t
+        {
+            Identity = 0,
+            Zero,
+            One,
+            R,
+            G,
+            B,
+            A,
+        };
+
+        ComponentSwizzle() = default;
+        ComponentSwizzle(const Value& swizzle) : value(swizzle) {};
+        ComponentSwizzle(const uint32_t swizzle) { value = static_cast<Value>(swizzle); }
+
+        operator Value() const { return value; }
+        ComponentSwizzle& operator=(uint32_t op) { value = static_cast<Value>(op); return *this; }
+        constexpr uint32_t operator-(ComponentSwizzle::Value op) const { return value - op; }
+        constexpr bool operator!=(ComponentSwizzle::Value op) const { return value != op; }
+        constexpr bool operator==(ComponentSwizzle::Value op) const { return value == op; }
+
+    private:
+        Value value;
+    };
+
+    // General YCbCr sample info
+    struct YCbCrSampleInfo
+    {
+        Type*              pResultTy;
+        uint32_t           dim;
+        uint32_t           flags;
+        Value*             pImageDesc;
+        Value*             pSamplerDesc;
+        ArrayRef<Value*>   address;
+        const Twine&       instName;
```

Comment:
Is it possible to use an `llvm::Twine` as a member here? I am mostly worried about dangling pointers here with `YCbrCrSampleInfo` that outlive temporaries.
See http://llvm.org/docs/ProgrammersManual.html#the-twine-class and https://llvm.org/doxygen/classllvm_1_1Twine.html#details.